tests/support/plugins/file_server.py

```diff
@@ -59,7 +59,9 @@ class HtmlOnlyHandler(BaseHTTPRequestHandler):
     """Http handler."""
     def do_GET(self) -> None:
         """GET method handler."""
-        path = self.path[1:].split("?")[0]
+        path = self.path.split("?")[0]
+        if path.startswith("//"):
+            path = path[1:]
         try:
             with open(os.path.join(HTML_ROOT, path), mode="rb") as f:  # lgtm [py/path-injection]
```

Comment:
The difference between 3.10.5 and 3.10.6 is that `self.path` starts with `//` (as originally formatted by `file_server.where_is(url)`) in 3.10.5, whereas it's being normalized to `/` in 3.10.6. I made it so that this works in both.

Reply:
Also, this logic doesn't make sense as `HTML_ROOT` and `path` are absolute paths and worse, `HTML_ROOT` is no root for anything (it points to plugins), and in the end is being ignored by `os.path.join()`.

Reply:
There are _very_ old tests, if there are ideas for improvements for later, please just make an issue to record them.

Reply:
I redesigned this to avoid absolute paths in URLs, making it more robust and safer.